src/UNL/MediaHub/Manager/PostHandler.php

```diff
@@ -515,6 +550,134 @@ function handleDeleteMedia()
 
         UNL_MediaHub::redirect(UNL_MediaHub_Manager::getURL());
     }
+    
+    function handleAmara()
+    {
+        $media = UNL_MediaHub_Media::getById($this->post['media_id']);
+        
+        if (!$media) {
+            throw new Exception('Unable to find media', 404);
+        }
+
+        if (!$media->userHasPermission(UNL_MediaHub_AuthService::getInstance()->getUser(), UNL_MediaHub_Permission::USER_CAN_UPDATE)) {
+            throw new Exception('You do not have permission to edit this media.', 403);
+        }
+        
+        $result = $media->updateAmaraCaptions();
+        
+        if (!$result) {
+            //No tracks were found, fail early
+            $notice = new UNL_MediaHub_Notice(
+                'Error',
+                'No amara captions could be found for this media',
+                UNL_MediaHub_Notice::TYPE_ERROR
+            );
+            UNL_MediaHub_Manager::addNotice($notice);
+            UNL_MediaHub::redirect($media->getEditCaptionsURL());
+
+            return;
+        }
+
+        $notice = new UNL_MediaHub_Notice(
+            'Success',
+            'The latest amara captions have been pulled.',
+            UNL_MediaHub_Notice::TYPE_SUCCESS
+        );
+        UNL_MediaHub_Manager::addNotice($notice);
+
+        UNL_MediaHub::redirect($media->getEditCaptionsURL());
+    }
+    
+    function handleRev()
+    {
+        $media = UNL_MediaHub_Media::getById($this->post['media_id']);
+
+        if (!$media) {
+            throw new Exception('Unable to find media', 404);
+        }
+
+        $user = UNL_MediaHub_AuthService::getInstance()->getUser();
+        
+        if (!$media->userHasPermission($user, UNL_MediaHub_Permission::USER_CAN_UPDATE)) {
+            throw new Exception('You do not have permission to edit this media.', 403);
+        }
+        
+        if (!isset($this->post['cost_object']) || empty($this->post['cost_object'])) {
+            throw new Exception('A cost object number must be provided', 400);
+        }
+        
+        $sanitized_co = str_replace(array('-', '.', '/'), '', $this->post['cost_object']);
```

Comment:
I'd do something like `preq_replace('/[^\d]/', '', $this->post['cost_object'])` to remove all non-digit characters and use that in your validation step above.